frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -575,6 +594,7 @@ export const applyVisualizationSettingsDataTransformations = (
   dataset: ChartDataset,
   xAxisModel: XAxisModel,
   seriesModels: SeriesModel[],
+  groupedSeriesKeys: DataKey[],
```

Comment:
I have an assumption that if we initially create the set of seriesModels with the grouped one, it may be easier and we will end up with the simpler code. Did you consider that approach?

Reply:
I considered it but it's not really feasible since we need the dataset in order to know which series to group, since we rank the series based on the values in the last datum. It would be a lot more complicated to try to pre-compute the last datum as opposed to this approach where we just have one additional parameter that we use transitively to help transform the dataset.